crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -1183,6 +1183,23 @@ impl CheckpointBuilder {
         let mut all_tx_digests =
             Vec::with_capacity(new_checkpoints.iter().map(|(_, c)| c.size()).sum());
 
+        // When upgrading to a data-quarantining build, we need to persist the transactions and effects
+        // to the database for crash recovery. After the upgrade this is no longer needed, because recovery
+        // is driven by replay of consensus commits.
+        // This only updates the content-addressed stores (similar to state sync), it does not mark any
+        // transactions as executed.
```

Comment:
maybe also a
`    /// TODO: remove this once all nodes have upgraded to data-quarantining.`
here as well?

Reply:
i could, although i already checked in a revert of this entire commit in my DQ branch so I'm pretty sure i won't forget to remove this